breakout_1.md

```diff
@@ -0,0 +1,182 @@
+```
+$ cd PATH/TO/FOLDER/
+```
+Clone repository from github.com to your computer. [URL] is the URL for the repository of the project leader and will
+include their username. It should be like `https://github.com/USERNAME/ASB_Tutorial`.
+```
+$ git clone [URL]
+```
+You now have a copy of the repository located at `PATH/TO/FOLDER/`. Go check it out! You'll see all the files that are 
+present on the main page of the repository on GitHub. Now you need to switch to your branch before making any changes to 
+files:
+```
+$ git checkout [BRANCH-NAME]
+```
+Now you're ready to fix some bugs!
+
+## 2. Making changes to files
+#### a. Debug subscript 
+This isn't a tutorial on debugging, so the bugs are easy to fix and solutions are commented out at the bottom of each 
+subscript. Open MATLAB (or a basic text editor) and fix the bug in your subscript. Save the debugged file (with the 
+same filename).
+#### b. Commit changes
+You made changes to files in the repository and want these changes to be tracked by Git. Git takes "snapshots" called 
+"commits" of your repository files, but you have to tell it 1) which files to include in the commit, 2) when to make
+the commit and include a message describing the changes in the commit, and 3) update the remote repository on GitHub 
+with your local changes. These three steps are performed in Git Bash and are outlined below.
+
+##### 1. Tell Git that files have been changed and should be included in the commit. 
+Git will compare the current state of the files to their previous state and identify any changes made. Files that have
+been changed will be "staged". In the "snapshot" analogy, this step is like wrangling your family members right before 
+taking the picture.
+```
+$ git add .
+```
+View the files Git has identified as undergoing some changes (lines starting with `>>>` represent a message returned 
+upon executing `git status`):
+```
+$ git status
+
+>>> On branch [BRANCH-NAME] 
+>>> Your branch is up to date with 'origin/[BRANCH-NAME]'.
+>>>
+>>> Changes to be commited:
+>>>   (use "git reset HEAD <file>..." to unstage)
+>>>       modified: script_1.m
+```
+This returned message tells you a few things:
+1. The branch you're working on.
+2. It found changes that are staged (the result of `git add .`)
+3. How to remove a specific file from being staged (`git reset HEAD [FILENAME]`)
+4. Which file was changed. 
+
+##### 2. Tell Git when to make the commit
+To continue using the "snapshot" analogy, you need to decide when to take the "snapshot". In this case, you fixed a bug 
+and this represents a meaningful level of changes made to your script. To help organize commits, you need to add a message
+that will be associated with the staged changes. For now, make it short and sweet. For more information on meaningful 
+commit messages, read [this blog post](https://chris.beams.io/posts/git-commit/) later. Remember: Git will only commit the 
+*staged* changes.
+```
```

Comment:
I  don't really like the 'when to make the commit'. You can't chose a time that it commits, it'll just commit once you hit enter.

Also, git add really takes the snapshot, git commit doesn't really take a snapshot, the commit is the tracking mechanism for the snapshot.

Reply:
I understand your point about not being able to "schedule" commits, but git only takes a snapshot when you tell it to. I want to make it clear that you have to decide for yourself when to commit (it's not continuously happening). 

And nah man, `git add` is moving files into the staging area. `git commit` is the snapshot. The Git manual literally uses "snapshot" and "commit" interchangeably: https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository

Reply:
Aight, Well i hate that wording lol